File: ccbusDemo/server/ccbusdemo/src/main/java/ccbus/demo/ccbus/web/components/EcmaInner.java
package ccbus.demo.ccbus.web.components;

import ccbus.connect.core.ccbus.ExtComponent;
import ccbus.connect.system.Prop;
import ccbus.connect.system.PropEnd;
import ccbus.connect.system.State;
import ccbus.connect.system.StateEnd;

import java.util.function.Function;

public class EcmaInner extends ExtComponent {
    @Prop
    String propName="";
    @PropEnd

    @State
    String stateName="";
    @StateEnd

    public Function<Integer,Integer> handleOnChangeD=(Integer b) -> {return b+1;};

    public EcmaInner() {
        super();
        onChangeProp("propName",(Object prop,Object propPrev) ->
        {
            this.stateKey("stateName",prop);
            return 0;
        });
    }
}

File: ccbusDemo/client/ccbusdemo/src/components/EcmaInner.js
import {ExtComponent} from 'ccbus';
import React from 'react';
import {PropTypes} from 'prop-types';

export class EcmaInner extends ExtComponent
{
    
    /**
    * @returns {}
    */ 
    constructor()
    {
        super();
        this.onChangeProp("propName",(prop,propPrev) => 
        {
            this.stateKey("stateName",prop);
            return 0;
        });
    }
    
    /**
    * @type {Object}
    */ 
    state = {
        
        /**
        * @type {String}
        */ 
        stateName: ""
    };
    static 
    /**
    * @type {Object}
    */ 
    propTypes = {
        
        /**
        * @type {String}
        */ 
        propName: PropTypes.string
    };
    static 
    /**
    * @type {Object}
    */ 
    defaultProps = {
        
        /**
        * @type {String}
        */ 
        propName: ""
    };
    
    /**
    * @type {Function<Integer,Integer>}
    */ 
    handleOnChangeD = (b) => 
    {
        return b+1;
    };
    
    render()
    {
        return <div>Inner says: Prop:{this.props.propName}<br/>State:{this.state.stateName}</div>
    }
}